package/gluon-neighbour-info/src/gluon-neighbour-info.c

```diff
@@ -69,8 +72,22 @@ void tv_subtract (struct timeval *r, const struct timeval *a, const struct timev
 	}
 }
 
-ssize_t recvtimeout(int socket, void *buffer, size_t length, int flags, const struct timeval *timeout) {
+void resize_recvbuffer(size_t recvlen)
+{
+	free(recvbuffer);
+	recvbuffer = malloc(recvlen);
```

Comment:
Any reason not to use `realloc` here?

Reply:
If there is no data to be retained in the buffer, `realloc` would just caused an unnecessary memcpy if the buffer needs to be relocated.